import os
os.environ['TF_CPP_MIN_LOG_LEVEL'] = '2'
import tensorflow as tf
from tensorflow import keras
from tensorflow.keras import layers
import numpy as np
np.set_printoptions(suppress=True,threshold=np.inf,formatter={'float_kind':'{:16.7f}'.format})
tf.keras.backend.set_floatx('float64')
in0Max51485 = tf.keras.layers.Input(shape=([1, 2]))
in1Max51485 = tf.keras.layers.Input(shape=([1, 2]))

Max51485 = keras.layers.Maximum(name = 'Max51485', )([in0Max51485,in1Max51485])
Glo16832 = keras.layers.GlobalMaxPool2D(name = 'Glo16832', )(Max51485)
Sof31953 = keras.layers.Softmax(axis=1, name = 'Sof31953', )(Glo16832)
model = tf.keras.models.Model(inputs=[in0Max51485,in1Max51485], outputs=Sof31953)
in0Max51485 = tf.constant([[[0.8599, 0.8947]]])
in1Max51485 = tf.constant([[[0.9298, 0.2445]]])
print (np.array2string(model.predict([in0Max51485,in1Max51485],steps=1), separator=', '))


LMax51485 = maximum_layer([[[[0.8599, 0.8947]]], [[[0.9298, 0.2445]]]], Max51485), 
LGlo16832 = global_max_pool2D_layer(Max51485, Glo16832), 
LSof31953 = softmax_layer(Glo16832, 1, Sof31953), 
exec_layers([LMax51485,LGlo16832,LSof31953],["Max51485","Glo16832","Sof31953"],Sof31953,"Sof31953")

Actual (Unparsed): 
 ValueError('Input ' + str(input_index) + ' of layer ' +ValueError: Input 0 of layer Glo16832 is incompatible with the layer: expected ndim=4, found ndim=3. Full shape received: (None, 1, 2)

Expected (Unparsed): 
Glo16832: Dimension error, Input Shape [1,1,2]

Actual:   

Expected: 